### PR TITLE
Home 3.1

### PR DESCRIPTION
**Pull request by YaroslavKSE**

### CODE REVIEW

File: Home/Home 3.1/CalculateInverseOrAlternative/Tests/CalculateInverseOrAlternativeTests.swift [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+@testable import CalculateInverseOrAlternative
+
+class CalculateInverseOrAlternativeTests: XCTestCase {
+
+    func testInverseOfProduct() {
+        // Arrange and Act
+        let result = calculateInverseOrAlternative(x: 2.0, y: 3.0, z: 4.0)
+
+        // Assert
+        XCTAssertEqual(result, 1.0 / 24.0, accuracy: 1e-9)
```
Comment (vovaskochko) [inline Home/Home 3.1/CalculateInverseOrAlternative/Tests/CalculateInverseOrAlternativeTests.swift]: as we are using TDD then we should perform that refactoring step after make unit test green. And that refactoring includes unit test code as well. in this case we have duplication of accuracy value. You can make it a constant member of this test class doubleAccuracy or something like that and reuse it

File: Home/Home 3.1/CalculateInverseOrAlternative/Tests/CalculateInverseOrAlternativeTests.swift [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+        // Assert
+        XCTAssertEqual(result, 1.0, accuracy: 1e-9)
+    }
+
+    func testMixedCase() {
+        // Arrange and Act
+        let result = calculateInverseOrAlternative(x: 1.0, y: 0.0, z: -1.0)
+
+        // Assert
+        XCTAssertEqual(result, 1.0 + (0.0 + 1.0) * (-1.0 - 1.0), accuracy: 1e-9)
```
Comment (vovaskochko) [inline Home/Home 3.1/CalculateInverseOrAlternative/Tests/CalculateInverseOrAlternativeTests.swift]: just calculate the result of formula -2.0 as it doesn't matter for test how we got the result the idea is to specify expected result.

**Review by vovaskochko — APPROVED** [review]

File: Home/Home 3.1/CalculateInverseOrAlternative/Tests/CalculateInverseOrAlternativeTests.swift (hunk 2 above)
Comment (YaroslavKSE) [reply Home/Home 3.1/CalculateInverseOrAlternative/Tests/CalculateInverseOrAlternativeTests.swift]: Refactored in [c37c93d](https://github.com/YaroslavKSE/Professinal-Programming/pull/8/commits/c37c93dd5a460c3a0e332d7076fd6e5df7dd0284)

File: Home/Home 3.1/CalculateInverseOrAlternative/Tests/CalculateInverseOrAlternativeTests.swift (hunk 1 above)
Comment (YaroslavKSE) [reply Home/Home 3.1/CalculateInverseOrAlternative/Tests/CalculateInverseOrAlternativeTests.swift]: Refactored in [c37c93d](https://github.com/YaroslavKSE/Professinal-Programming/pull/8/commits/c37c93dd5a460c3a0e332d7076fd6e5df7dd0284)

**Pull request merged**